### PR TITLE
Add python 3 to the dependency list and fix several build warnings

### PR DESCRIPTION
**Pull request by vdogaru**

### CODE REVIEW

**Comment by ddebrunner** [comment]
Can one of the admins verify this patch?

File: test/java/src/com/ibm/streamsx/topology/test/api/LowLatencyTest.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ public void threadedPortTest() throws Exception{
                 lowLatencyTag = (String) placement.get(JOperator.PLACEMENT_LOW_LATENCY_REGION_ID);
             String kind = (String)op.get("kind");
             JSONObject queue = (JSONObject) op.get("queue");
-            if(queue != null && (lowLatencyTag!=null || lowLatencyTag.equals(""))){
+            if(queue != null && (lowLatencyTag == null || lowLatencyTag.equals(""))){
```
Comment (ddebrunner) [inline test/java/src/com/ibm/streamsx/topology/test/api/LowLatencyTest.java]: I assume this was changed to a code not reached warning, but not sure if the replacement makes much sense either.

**Comment by ddebrunner** [comment]
test now

**Comment by ddebrunner** [comment]
test now

File: test/java/src/com/ibm/streamsx/topology/test/api/LowLatencyTest.java (hunk 1 above)
Comment (vdogaru) [inline test/java/src/com/ibm/streamsx/topology/test/api/LowLatencyTest.java]: Replacing `equals("")` with i`sEmpty()`:

1. `if (queue != null && (lowLatencyTag == null || lowLatencyTag.isEmpty()))`
2. `if (queue != null && (lowLatencyTag != null && !lowLatencyTag.isEmpty()))`

I'm not familiar with the graph model and cannot decide: which alternative makes more sense, (1) or (2)?

**Pull request merged**